packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuManager.kt

```diff
@@ -212,13 +212,13 @@ object DevMenuManager : DevMenuManagerInterface, LifecycleEventListener {
         ?: reactContext.applicationContext as Application)
     maybeStartDetectors(devMenuHost.getContext())
 
-    settings = testInterceptor.overrideSettings()
+    settings = (testInterceptor.overrideSettings()
         ?: if (reactContext.hasNativeModule(DevMenuSettings::class.java)) {
           reactContext.getNativeModule(DevMenuSettings::class.java)!!
         } else {
           DevMenuDefaultSettings()
-        }.also {
-          shouldLaunchDevMenuOnStart = it.showsAtLaunch || !it.isOnboardingFinished
+        }).also {
+          shouldLaunchDevMenuOnStart = shouldAutoLaunch && (it.showsAtLaunch || !it.isOnboardingFinished)
```

Comment:
I find the name of this variable a bit confusing: `shouldAutoLaunch` sounds like it would force the menu to launch. It's more or less synonymous to `shouldLaunchDevMenuOnStart`. It's also similar to `showsAtLaunch`.

Perhaps it should be `canLaunchDevMenuOnStart `? This seems clearer:
```suggestion
          shouldLaunchDevMenuOnStart = canLaunchDevMenuOnStart && (it.showsAtLaunch || !it.isOnboardingFinished)
```
Possibly `DevLauncherController.kt` could use the same name to make it easier to search for.